tendrl/node_agent/persistence/disk.py

```diff
@@ -0,0 +1,39 @@
+from tendrl.common.etcdobj.etcdobj import EtcdObj
+from tendrl.common.etcdobj import fields
+
+
+class Disk(EtcdObj):
+    """A table of the Disk, lazily updated
+
+    """
+    __name__ = 'nodes/%s/Disks/%s/%s'
+
+    node_id = fields.StrField("node_id")
+    device_name = fields.StrField("device_name")
+    fs_type = fields.StrField("fs_type")
+    fs_uuid = fields.StrField("fs_uuid")
+    model = fields.StrField("model")
+    mount_point = fields.StrField("mount_point")
+    name = fields.StrField("name")
+    parent = fields.StrField("parent")
+    size = fields.StrField("size")
+    disk_type = fields.StrField("type")
+    used = fields.StrField("used")
+    ssd = fields.StrField("ssd")
+    vendor = fields.StrField("vendor")
+    disk_id = fields.StrField("disk_id")
+
+    def render(self):
+        if self.used == "True":
+            self.__name__ = self.__name__ % (
+                self.node_id,
+                "used",
+                self.disk_id
+            )
+        else:
+            self.__name__ = self.__name__ % (
+                self.node_id,
+                "free",
```

Comment:
I am storing free nodes separately and used node separately
free nodes comes under : nodes/node_id/Disks/free/disk_name
 used nodes comes under : nodes/node_id/Disks/used/disk_id